gocd/gocd.go

```diff
@@ -0,0 +1,210 @@
+			Tabs       []interface{} `json:"tabs"`
+			Artifacts  []interface{} `json:"artifacts"`
+			Properties interface{}   `json:"properties"`
+		} `json:"jobs"`
+	} `json:"stages"`
+	TrackingTool interface{} `json:"tracking_tool"`
+	Timer        interface{} `json:"timer"`
+}
+
+func unmarshalPipeline(data []byte) (pipeline Pipeline, err error) {
+	err = json.Unmarshal(data, &pipeline)
+	if err != nil {
+		return pipeline, err
+	}
+	return pipeline, nil
+}
+
+func unmarshalPipelineConfig(data []byte) (pipelineConfig PipelineConfig, err error) {
+	err = json.Unmarshal(data, &pipelineConfig)
+	if err != nil {
+		return pipelineConfig, err
+	}
+	return pipelineConfig, nil
+}
+
+func ReadPipelineJSONFromFile(path string) (pipeline Pipeline, err error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return pipeline, err
+	}
+	return unmarshalPipeline(data)
+}
+
+func pipelineConfigPOST(gocdULR string, pipelineConfig PipelineConfig) (pipeline Pipeline, err error) {
+
+	// Generated by curl-to-Go: https://mholt.github.io/curl-to-go
+
+	payloadBytes, err := json.Marshal(pipelineConfig)
+	if err != nil {
+		return pipeline, err
+	}
+	payloadBody := bytes.NewReader(payloadBytes)
+
+	req, err := http.NewRequest("POST", gocdULR+"/go/api/admin/pipelines", payloadBody)
+	if err != nil {
+		return pipeline, err
+	}
+	user := os.Getenv("GOCDUSER")
+	pass := os.Getenv("GOCDPASSWORD")
+	req.SetBasicAuth(user, pass)
+	req.Header.Set("Accept", "application/vnd.go.cd.v1+json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return pipeline, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return pipeline, err
+	}
+
+	var prettyJSON bytes.Buffer
+	err = json.Indent(&prettyJSON, body, "", "\t")
+	if err != nil {
+		return pipeline, err
+	}
+	//fmt.Println("pipelineConfigPOST:", string(prettyJSON.Bytes()))
+
+	return unmarshalPipeline(body)
+}
+
+func pipelineGET(gocdULR string, pipelineName string) (pipeline Pipeline, err error) {
+
+	// Generated by curl-to-Go: https://mholt.github.io/curl-to-go
+
+	req, err := http.NewRequest("GET", gocdULR+"/go/api/admin/pipelines/"+pipelineName, nil)
+	if err != nil {
+		return pipeline, err
+	}
+	user := os.Getenv("GOCDUSER")
+	pass := os.Getenv("GOCDPASSWORD")
+	req.SetBasicAuth(user, pass)
+	req.Header.Set("Accept", "application/vnd.go.cd.v1+json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return pipeline, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return pipeline, err
+	}
+
+	var prettyJSON bytes.Buffer
+	err = json.Indent(&prettyJSON, body, "", "\t")
+	if err != nil {
+		return pipeline, err
+	}
+	//fmt.Println("pipelineConfigGET:", string(prettyJSON.Bytes()))
+
+	return unmarshalPipeline(body)
+}
+
+func Push(gocdULR string, path string, group string) {
```

Comment:
Go likes to have all public functions contain a line of comment, starting with the function name, like // Push ....
We should follow that convention.